src/FSharpx.Core/ComputationExpressions/Monad.fs

```diff
@@ -61,6 +61,12 @@ module Async =
     /// Right-to-left Kleisli composition
     let inline (<=<) x = flip (>=>) x
 
+    let attempt m = async { 
+                             try  let! outp = m
+                                  return Choice1Of2 <| outp
+                             with exn -> return Choice2Of2 exn
+                    }
+
```

Comment:
This function should be moved to [`FSharpx.Async`](https://github.com/fsprojects/FSharpx.Async) as an extension to `Async`.